File: WebShop/ClientApp/src/models/Product.ts
import Image from './Image'
import PriceQuote from './PriceQuote'
import ProductSpecification from './ProductSpecification'
export default class Product {
    id?: number
    name?: string
    description?: string
    images?: Array<Image>
    prices?: Array<PriceQuote>
    itemNumber?: string
    specifications?: Array<ProductSpecification>
}

File: WebShop/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace WebShop.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public IEnumerable<Image> Images { get; set; }
        public IEnumerable<PriceQuote> Prices { get; set; }
        public string ItemNumber { get; set; }

        public IEnumerable<ProductSpecification> Specifications { get; set; }
    }
}
